Metabase query INFORMATION_SCHEMA.TABLES failed

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)

SELECT TABLE_SCHEMA TABLE_CAT, NULL TABLE_SCHEM, TABLE_NAME, IF(TABLE_TYPE='BASE TABLE' or TABLE_TYPE='SYSTEM VERSIONED', 'TABLE', TABLE_TYPE) as TABLE_TYPE, TABLE_COMMENT REMARKS, NULL TYPE_CAT, NULL TYPE_SCHEM, NULL TYPE_NAME, NULL SELF_REFERENCING_COL_NAME, NULL REF_GENERATION FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = database() AND  TABLE_TYPE IN ('BASE TABLE','SYSTEM VERSIONED','PARTITIONED TABLE' ,'VIEW','FOREIGN TABLE','MATERIALIZED VIEW','EXTERNAL TABLE') ORDER BY TABLE_TYPE, TABLE_SCHEMA, TABLE_NAME


### Expected behavior (Required)
QueryOK
### Real behavior (Required)
2022-09-19 09:54:27,275 INFO sync.util :: STARTING: step 'sync-metabase-metadata' for mysql Database 2 'starrocks'
2022-09-19 09:54:27,282 WARN sync.util :: Error syncing _metabase_metadata table for mysql Database 2 'starrocks'
java.sql.SQLTransientConnectionException: (conn=5) com.starrocks.sql.optimizer.operator.scalar.CallOperator cannot be cast to com.starrocks.sql.optimizer.operator.scalar.ConstantOperator
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:362)
	at org.mariadb.jdbc.MariaDbStatement.executeQuery(MariaDbStatement.java:611)
	at org.mariadb.jdbc.MariaDbDatabaseMetaData.executeQuery(MariaDbDatabaseMetaData.java:497)
	at org.mariadb.jdbc.MariaDbDatabaseMetaData.getTables(MariaDbDatabaseMetaData.java:685)
	at com.mchange.v2.c3p0.impl.NewProxyDatabaseMetaData.getTables(NewProxyDatabaseMetaData.java:2938)
	at metabase.driver.sql_jdbc.sync.describe_database$db_tables$fn__60321.invoke(describe_database.clj:88)
	at metabase.driver.sql_jdbc.sync.common$reducible_results$reify__59813.reduce(common.clj:34)
	at clojure.core$transduce.invokeStatic(core.clj:6946)
	at clojure.core.Eduction.reduce(core.clj:7750)
	at clojure.core$reduce.invokeStatic(core.clj:6885)
	at clojure.core$into.invokeStatic(core.clj:6958)
	at clojure.core$into.invoke(core.clj:6950)
	at metabase.driver.sql_jdbc.sync.describe_database$describe_database$fn__60355$default_active_tbl_fn__60356.invoke(describe_database.clj:153)
	at metabase.driver.sql_jdbc.sync.describe_database$describe_database$fn__60355.invoke(describe_database.clj:162)
	at metabase.driver.sql_jdbc.sync.describe_database$describe_database.invokeStatic(describe_database.clj:146)
	at metabase.driver.sql_jdbc.sync.describe_database$describe_database.invoke(describe_database.clj:143)
	at metabase.driver.sql_jdbc$fn__85198.invokeStatic(sql_jdbc.clj:62)
	at metabase.driver.sql_jdbc$fn__85198.invoke(sql_jdbc.clj:60)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.sync.fetch_metadata$fn__66047$db_metadata__66052$fn__66053.invoke(fetch_metadata.clj:14)
	at metabase.sync.fetch_metadata$fn__66047$db_metadata__66052.invoke(fetch_metadata.clj:11)
	at metabase.sync.sync_metadata.metabase_metadata$fn__67024$sync_metabase_metadata_BANG___67029$fn__67030$fn__67031.invoke(metabase_metadata.clj:100)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:158)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:151)
	at metabase.sync.sync_metadata.metabase_metadata$fn__67024$sync_metabase_metadata_BANG___67029$fn__67030.invoke(metabase_metadata.clj:93)
	at metabase.sync.sync_metadata.metabase_metadata$fn__67024$sync_metabase_metadata_BANG___67029.invoke(metabase_metadata.clj:88)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.sync.util$fn__42311$run_step_with_metadata__42316$fn__42320$fn__42322.doInvoke(util.clj:388)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:100)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:94)
	at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:117)
	at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:114)
	at metabase.sync.util$fn__42311$run_step_with_metadata__42316$fn__42320.invoke(util.clj:383)
	at metabase.sync.util$fn__42311$run_step_with_metadata__42316.invoke(util.clj:378)
	at metabase.sync.util$fn__42530$run_sync_operation__42535$fn__42536$fn__42544.invoke(util.clj:493)
	at metabase.sync.util$fn__42530$run_sync_operation__42535$fn__42536.invoke(util.clj:491)
	at metabase.sync.util$fn__42530$run_sync_operation__42535.invoke(util.clj:485)
	at metabase.sync.sync_metadata$fn__68442$sync_db_metadata_BANG___68447$fn__68448$fn__68449.invoke(sync_metadata.clj:50)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:158)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:151)
	at clojure.core$partial$fn__5910.invoke(core.clj:2647)
	at metabase.driver$fn__33070.invokeStatic(driver.clj:599)
	at metabase.driver$fn__33070.invoke(driver.clj:599)
	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
	at metabase.sync.util$sync_in_context$fn__42220.invoke(util.clj:136)
	at metabase.sync.util$with_db_logging_disabled$fn__42217.invoke(util.clj:127)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:100)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:94)
	at metabase.sync.util$with_start_and_finish_logging$fn__42206.invoke(util.clj:112)
	at metabase.sync.util$with_sync_events$fn__42201.invoke(util.clj:86)
	at metabase.sync.util$with_duplicate_ops_prevented$fn__42192.invoke(util.clj:65)
	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:179)
	at metabase.sync.util$do_sync_operation.invoke(util.clj:176)
	at metabase.sync.sync_metadata$fn__68442$sync_db_metadata_BANG___68447$fn__68448.invoke(sync_metadata.clj:49)
	at metabase.sync.sync_metadata$fn__68442$sync_db_metadata_BANG___68447.invoke(sync_metadata.clj:46)
	at metabase.api.database$fn__76560$fn__76561.invoke(database.clj:832)
	at clojure.core$binding_conveyor_fn$fn__5823.invoke(core.clj:2047)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: com.starrocks.sql.optimizer.operator.scalar.CallOperator cannot be cast to com.starrocks.sql.optimizer.operator.scalar.ConstantOperator
	at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:194)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:262)
	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:356)
	... 65 more
Caused by: java.sql.SQLException: com.starrocks.sql.optimizer.operator.scalar.CallOperator cannot be cast to com.starrocks.sql.optimizer.operator.scalar.ConstantOperator
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1695)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1557)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1520)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:256)
	... 66 more
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.5

--- a/fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java
@@ -864,7 +864,8 @@ public class PlanFragmentBuilder {
                 if (predicate instanceof BinaryPredicateOperator) {
                     if (((BinaryPredicateOperator) predicate).getBinaryType() ==
                             BinaryPredicateOperator.BinaryType.EQ) {
-                        if (predicate.getChildren().get(0) instanceof ColumnRefOperator) {
+                        if (predicate.getChildren().get(0) instanceof ColumnRefOperator &&
+                                predicate.getChildren().get(1) instanceof ConstantOperator) {
                             ColumnRefOperator columnRefOperator = (ColumnRefOperator) predicate.getChildren().get(0);
                             ConstantOperator constantOperator = (ConstantOperator) predicate.getChildren().get(1);
                             switch (columnRefOperator.getName()) {

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/ScanTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/ScanTest.java
@@ -372,7 +372,7 @@ public class ScanTest extends PlanTestBase {
     }
 
     @Test
-    public void testMetabaseScan() throws Exception {
+    public void testSchemaScanWithWhereConstantFunction() throws Exception {
         String sql = "SELECT TABLE_SCHEMA TABLE_CAT, NULL TABLE_SCHEM, TABLE_NAME, " +
                 "IF(TABLE_TYPE='BASE TABLE' or TABLE_TYPE='SYSTEM VERSIONED', 'TABLE', TABLE_TYPE) as TABLE_TYPE, " +
                 "TABLE_COMMENT REMARKS, NULL TYPE_CAT, NULL TYPE_SCHEM, NULL TYPE_NAME, NULL SELF_REFERENCING_COL_NAME, " +

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/ScanTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/ScanTest.java
@@ -370,4 +370,15 @@ public class ScanTest extends PlanTestBase {
         Assert.assertTrue(((SchemaScanNode) plan.getScanNodes().get(0)).getSchemaDb().equals("information_schema"));
         Assert.assertTrue(((SchemaScanNode) plan.getScanNodes().get(0)).getSchemaTable().equals("columns"));
     }
+
+    @Test
+    public void testMetabaseScan() throws Exception {
+        String sql = "SELECT TABLE_SCHEMA TABLE_CAT, NULL TABLE_SCHEM, TABLE_NAME, " +
+                "IF(TABLE_TYPE='BASE TABLE' or TABLE_TYPE='SYSTEM VERSIONED', 'TABLE', TABLE_TYPE) as TABLE_TYPE, " +
+                "TABLE_COMMENT REMARKS, NULL TYPE_CAT, NULL TYPE_SCHEM, NULL TYPE_NAME, NULL SELF_REFERENCING_COL_NAME, " +
+                "NULL REF_GENERATION FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = database() AND  " +
+                "TABLE_TYPE IN ('BASE TABLE','SYSTEM VERSIONED','PARTITIONED TABLE','VIEW','FOREIGN TABLE'," +
+                "'MATERIALIZED VIEW','EXTERNAL TABLE') ORDER BY TABLE_TYPE, TABLE_SCHEMA, TABLE_NAME";
+        getExecPlan(sql);
+    }
 }

